speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SocketBlackBox.inl

```diff
@@ -49,10 +49,17 @@ void SocketBlackBox<BlackBoxData, PointerManager>::Initialize()
 /// @brief                          Sets keys and values for the functions that retrieve the correct information. Also initializes the AI
 /// @param p_initialBlackBoxData  The initial drive situation
 template <class BlackBoxData, class PointerManager>
-void SocketBlackBox<BlackBoxData, PointerManager>::Initialize(BlackBoxData& p_initialBlackBoxData, BlackBoxData* p_tests, int p_amountOfTests)
+void SocketBlackBox<BlackBoxData, PointerManager>::Initialize(bool p_connectAsync, BlackBoxData& p_initialBlackBoxData, BlackBoxData* p_tests, int p_amountOfTests)
 {
     Initialize();
-    m_server.ConnectAsync();
+    std::cout << "Connecting ";
+    if (p_connectAsync)
+    {
+        m_server.ConnectAsync();
+        m_asyncConnection = true;
+        std::cout << "a";
+    }
+    std::cout << "sync" << std::endl;
```

Comment:
I think we can always do a non-async connection, as doing connectasync() followed by awaitclientconnection is the same as just doing awaitclientconnection